nautobot_golden_config/utilities/config_postprocessing.py

```diff
@@ -0,0 +1,166 @@
+        # Only importing the encrypt helpers as complements to get_secrets filter
+        if name in ["encrypt_type5", "encrypt_type7"]:
+            jinja_env.filters[name] = func
+
+    # Wrapper for get_secret filter that includes user argument to ensure
+    # that secrets are only rendered by authorized users.
+    # To call this method, the view verifies that it's an authenticated request.
+    jinja_env.filters["get_secret_by_secret_group_slug"] = partial(get_secret_by_secret_group_slug, request.user)
+
+    try:
+        template = jinja_env.from_string(config_postprocessing)
+    except jinja_errors.TemplateAssertionError as error:
+        return f"Jinja encountered an TemplateAssertionError: '{error}'; check the template for correctness"
+
+    device_data = _get_device_agg_data(configs.device, request)
+
+    try:
+        return template.render(device_data)
+
+    except jinja_errors.UndefinedError as error:
+        raise RenderConfigToPushError(
+            f"Jinja encountered and UndefinedError: {error}, check the template for missing variable definitions.\n"
+        ) from error
+    except jinja_errors.TemplateSyntaxError as error:  # Also catches subclass of TemplateAssertionError
+        raise RenderConfigToPushError(
+            f"Jinja encountered a SyntaxError at line number {error.lineno},"
+            f"check the template for invalid Jinja syntax.\n"
+        ) from error
+    except jinja_errors.TemplateError as error:  # Catches all remaining Jinja errors
+        raise RenderConfigToPushError(
+            "Jinja encountered an unexpected TemplateError; check the template for correctness\n"
+        ) from error
+
+
+def get_config_postprocessing(configs: models.GoldenConfig, request: HttpRequest) -> str:
+    """Renders final configuration  artifact from intended configuration.
+
+    It chains multiple callables to transform an intended configuration into a configuration that can be pushed.
+    Each callable should match the following signature:
+    `my_callable_function(config_postprocessing: str, configs: models.GoldenConfig, request: HttpRequest)`
+
+    Args:
+        configs (models.GoldenConfig): Golden Config object per device, to retrieve device info, and related configs.
+        request (HttpRequest): HTTP request for context.
+    """
+    if not ENABLE_CONFIG_POSTPROCESSING:
+        return (
+            "Generation of intended configurations postprocessing it is not enabled, check your plugin configuration."
+        )
+
+    config_postprocessing = configs.intended_config
+    if not config_postprocessing:
+        return (
+            "No intended configuration is available. Before rendering the configuration with postprocessing, "
+            "you need to generate the intended configuration."
+        )
+
+    # Available functions to create the final intended configuration
+    default_config_postprocessing_callables = [render_secrets]
+    config_postprocessing_callable = PLUGIN_CFG.get(
```

Comment:
I am realizing that you are sending the actual callable now. I "think" the best approach is to convert to callable with the import string method. You can see how I implemented this within the fw models:

https://github.com/nautobot/nautobot-plugin-firewall-models/blob/75dddca0a48c1a2be4fb6370190f3374d2fd6dd0/nautobot_firewall_models/models/capirca_models.py#L64

The reason is, allowing this to be a configuration option that you can set via environment variables and it is a somewhat established pattern within the nautobot ecosystem.